framework/include/cppmicroservices/ListenerFunctors.h

```diff
@@ -97,18 +151,56 @@ namespace cppmicroservices {
    */
   typedef std::function<void(const FrameworkEvent&)> FrameworkListener;
 
-  template<class X>
-  ServiceListener ServiceListenerMemberFunctor(X* x, void (X::*memFn)(const ServiceEvent&))
-  { return std::bind(memFn, x, std::placeholders::_1); }
-
-  template<class X>
-  BundleListener BundleListenerMemberFunctor(X* x, void (X::*memFn)(const BundleEvent&))
-  { return std::bind(memFn, x, std::placeholders::_1); }
+  /**
+   * \ingroup MicroServices
+   * \ingroup gr_listeners
+   *
+   * A convenience function that binds the member function <code>callback</code> of
+   * an object of type <code>R</code> and returns a <code>ServiceListener</code> object.
+   * This object can then be passed into <code>AddServiceListener()</code>.
+   *
+   * @tparam R The type containing the member function.
+   * @param receiver The object of type R.
+   * @param callback The member function pointer.
+   * @returns a ServiceListener object.
+   */
+  template<class R>
+  ServiceListener ServiceListenerMemberFunctor(R* receiver, void (R::*callback)(const ServiceEvent&))
```

Comment:
Sure. I had two choices - either deprecate these or add documentation to these convenience functions. But, std::bind is straightforward enough that clients wouldn't need these convenient functions I guess.